sycl/test-e2e/ESIMD/esimd_test_utils.hpp

```diff
@@ -568,7 +568,7 @@ std::unique_ptr<T, USMDeleter> usm_malloc_shared(queue q, int n) {
   return std::move(res);
 }
 
-template <class T> static const char *type_name();
+template <class T> const char *type_name() { return typeid(T).name(); }
```

Comment:
If the function is defined, then likely the warning will not be issued.

Reply:
yes, but that 'static' keyword did not seem to be needed for this template function